docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,202 @@
+Async Programming in C#/VB
+==========================
+By `Phillip Carter`_
+
+From a developer's perspective, the C#/VB async model is some nice syntax that hides complexity and allows you to focus on writing methods without worrying about things like thread pools and callbacks.
+
+The core of this are the ``async`` and ``await`` keywords (``Async`` and ``Await`` in VB.NET), which you sprinkle over otherwise normal methods. 
```

Comment:
I think it's just "VB", not "VB.NET".

It's true that the methods are otherwise normal. That said, there are a few categories of good candidates for async/await. I/O bound is one and the other is work you want to get off the main thread (do in the background). There are some more niche scenarios, too.